src/NHSD.BuyingCatalogue.Ordering.Services/OrderService.cs

```diff
@@ -0,0 +1,106 @@
+
+namespace NHSD.BuyingCatalogue.Ordering.Services
+{
+    public sealed class OrderService : IOrderService
+    {
+        private readonly ApplicationDbContext context;
+
+        public OrderService(ApplicationDbContext context)
+        {
+            this.context = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        public async Task<Order> GetOrder(CallOffId callOffId)
+        {
+            return await context.Order
+                .Where(o => o.Id == callOffId.Id)
+                .Include(o => o.OrderingParty).ThenInclude(p => p.Address)
+                .Include(o => o.OrderingPartyContact)
+                .Include(o => o.Supplier).ThenInclude(s => s.Address)
+                .Include(o => o.SupplierContact)
+                .Include(o => o.ServiceInstanceItems).Include(o => o.OrderItems).ThenInclude(i => i.CatalogueItem)
+                .Include(o => o.OrderItems).ThenInclude(i => i.OrderItemRecipients).ThenInclude(r => r.Recipient)
+                .Include(o => o.OrderItems).ThenInclude(i => i.PricingUnit)
+                .AsNoTracking()
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task<IList<Order>> GetOrderList(Guid organisationId)
+        {
+            return await context.OrderingParty
+                .Where(o => o.Id == organisationId)
+                .SelectMany(o => o.Orders)
+                .AsNoTracking()
+                .ToListAsync();
+        }
+
+        public async Task<Order> GetOrderSummary(CallOffId callOffId)
+        {
+            return await context.Order
+                .Where(o => o.Id == callOffId.Id)
+                .Include(o => o.OrderingParty)
+                .Include(o => o.OrderingPartyContact)
+                .Include(o => o.SupplierContact)
+                .Include(o => o.SelectedServiceRecipients)
+                .Include(o => o.OrderItems).ThenInclude(i => i.CatalogueItem)
+                .Include(o => o.OrderItems).ThenInclude(i => i.OrderItemRecipients)
+                .Include(o => o.Progress)
+                .AsQueryable()
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task<Order> GetOrderCompletedStatus(CallOffId callOffId)
+        {
+            return await context.Order
+                .Where(o => o.Id == callOffId.Id)
+                .Include(o => o.OrderingParty)
+                .Include(o => o.Supplier)
+                .Include(o => o.OrderItems).ThenInclude(i => i.CatalogueItem)
+                .Include(o => o.OrderItems).ThenInclude(i => i.OrderItemRecipients).ThenInclude(r => r.Recipient)
+                .Include(o => o.OrderItems).ThenInclude(i => i.PricingUnit)
+                .Include(o => o.Progress)
+                .SingleOrDefaultAsync();
+        }
+
+        [ItemCanBeNull]
+        public async Task<OrderingParty> GetOrderingParty(Guid organisationId)
+        {
+            return await context.OrderingParty.FindAsync(organisationId);
+        }
```

Comment:
Once it's been removed from the interface definition we can make this method `private`. Do we need the `[ItemCanBeNull]` annotation?

Reply:
Made method `GetOrderingParty` private and removed  `[ItemCanBeNull]`